### PR TITLE
v3: tweak types for location hooks

### PR DESCRIPTION
**Pull request by jeetiss**
fix https://github.com/molefrog/wouter/issues/366

### CODE REVIEW

**Review by AbdullahWins — COMMENTED** [review]
Good Job!

File: packages/wouter-preact/types/index.d.ts [hunk 1]
```diff
@@ -14,13 +14,14 @@ import {
...
-} from "./use-browser-location";
+} from "./location-hook";
+import { LocationHook } from "./use-browser-location";
 
 import { RouterObject, RouterOptions } from "./router";
 
 // re-export some types from these modules
-export { Path, BaseLocationHook, LocationHook } from "./use-browser-location";
+export { Path, BaseLocationHook } from "./location-hook";
+export { LocationHook } from "./use-browser-location";
```
Comment (jeetiss) [inline packages/wouter-preact/types/index.d.ts:24]: Should i remove this export?

Comment (molefrog) [reply packages/wouter-preact/types/index.d.ts:24]: Yes, and we can also rename `LocationHook` to `BrowserLocationHook` to be more specific.

**Review by molefrog — APPROVED** [review]
Nice!

I'll just write down breaking changes for the release notes:
- `LocationHook` type is no longer exported. Use `BaseLocationHook` for implementing custom location hooks or `BrowserLocationHook` exported from `"wouter/use-browser-location"` to get a more specific type.

**Pull request merged**